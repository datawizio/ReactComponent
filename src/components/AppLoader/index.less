@loader-bg-color: #ffffff;
@loader-color: rgb(142, 116, 198);
@loader-size: 110px;
@loader-space-size: 3px;

*,
*::before,
*::after {
  box-sizing: border-box;
}

.loader {
  background-color: @loader-bg-color;
  height: calc(100vh - 16px);
  width: 100%;
  display: flex;
  align-items: center;
  position: absolute;
  bottom: 0;
  right: 0;
  left: 0;
  top: 0;

  .loader-wrapper {
    position: relative;
    display: block;
    margin: auto;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    .loader-container {
      position: absolute;
      width: 100%;
      height: 100%;

      .loader-logo-container {
        position: relative;
        margin: 29px auto;
        width: @loader-size;
        height: @loader-size;
        padding: @loader-space-size;
        .loader-logo {
          width: @loader-size - @loader-space-size * 2;
          height: @loader-size - @loader-space-size * 2;
          background: url("/static/images/logo.svg");
        }
      }
    }

    .loader-spinner {
      margin: 25px auto;
      font-size: 10px;
      position: relative;
      text-indent: -9999em;
      border-top: 4px solid @loader-bg-color;
      border-right: 4px solid @loader-bg-color;
      border-bottom: 4px solid @loader-bg-color;
      border-left: 4px solid @loader-color;
      -webkit-transform: translateZ(0);
      -ms-transform: translateZ(0);
      transform: translateZ(0);
      -webkit-animation: loading 1.1s infinite linear;
      animation: loading 1.1s infinite linear;

      &,
      &:after {
        border-radius: 50%;
        width: @loader-size + 8;
        height: @loader-size + 8;
      }
    }
  }
}

@-webkit-keyframes loading {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes loading {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
