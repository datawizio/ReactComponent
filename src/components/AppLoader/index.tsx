import React from "react";
import clsx from "clsx";

import "./index.less";

export interface AppLoaderProps {
  /**
   * The source URL for the image
   * that will be displayed within the loader component.
   */
  imageSrc: string;
}

const AppLoader: React.FC<AppLoaderProps> = ({ imageSrc }) => {
  const theme = localStorage.getItem("theme");
  const isDark = theme === "dark";

  return (
    <div className={clsx("loader", isDark && "loader-dark")}>
      <div className="loader-wrapper">
        <div className="loader-container">
          <div className="loader-logo-container">
            <div className="loader-logo">
              <img alt="loader" src={imageSrc} />
            </div>
          </div>
        </div>
        <div className="loader-spinner"></div>
      </div>
    </div>
  );
};

export default AppLoader;
