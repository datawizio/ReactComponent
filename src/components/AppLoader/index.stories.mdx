import { Meta, Story, Preview, ArgsTable } from "@storybook/addon-docs/blocks";
import AppLoader from "./index";

<Meta title="Components/AppLoader" component={AppLoader} />

# AppLoader

The `AppLoader` component displays a loading indicator with an image, typically used while initializing the application.

<Preview>
  <Story name="Basic">
    <AppLoader imageSrc="https://bi.datawiz.io/static/images/logo.svg" />
  </Story>
</Preview>

## Props

<ArgsTable of={AppLoader} />
