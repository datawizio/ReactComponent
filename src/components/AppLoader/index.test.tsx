import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import AppLoader from "./index";

const mockProps = {
  imageSrc: "https://bi.datawiz.io/static/images/logo.svg"
};

const setUp = (props?: React.ComponentProps<typeof AppLoader>) => {
  return mount(<AppLoader {...props} />);
};

describe("AppLoader component", () => {
  let component: ReturnType<typeof setUp>;

  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("is rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("has the correct class", () => {
    expect(component.find(".loader").length).toBeTruthy();
  });

  it("has the correct image", () => {
    expect(component.find("img").props().src).toBe(mockProps.imageSrc);
  });
});
