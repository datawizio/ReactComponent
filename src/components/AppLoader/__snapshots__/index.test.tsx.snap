// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLoader component is rendered correctly 1`] = `
<AppLoader
  imageSrc="https://bi.datawiz.io/static/images/logo.svg"
>
  <div
    className="loader"
  >
    <div
      className="loader-wrapper"
    >
      <div
        className="loader-container"
      >
        <div
          className="loader-logo-container"
        >
          <div
            className="loader-logo"
          >
            <img
              alt="loader"
              src="https://bi.datawiz.io/static/images/logo.svg"
            />
          </div>
        </div>
      </div>
      <div
        className="loader-spinner"
      />
    </div>
  </div>
</AppLoader>
`;
